numpy/core/src/multiarray/descriptor.c

```diff
@@ -2391,7 +2398,10 @@ arraydescr_reduce(PyArray_Descr *self, PyObject *NPY_UNUSED(args))
         }
         obj = PyUString_FromFormat("%c%d",self->kind, elsize);
     }
-    PyTuple_SET_ITEM(ret, 1, Py_BuildValue("(Nii)", obj, 0, 1));
+    PyTuple_SET_ITEM(ret, 1, Py_BuildValue("(Nii)", obj, 0, !builtin));
```

Comment:
The third element of ret? It's set right at the end of the function, it contains the extra state information needed for arraydescr_setstate to rebuild the dtype (for non-builtin dtypes). The reduce function is allowed to just return a 2-tuple if the third element isn't needed.